packages/grid/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -670,4 +674,11 @@ export interface DataGridPropsWithoutDefaultValue extends CommonProps {
    * For each feature, if the flag is not explicitly set to `true`, the feature will be fully disabled and any property / method call will not have any effect.
    */
   experimentalFeatures?: Partial<GridExperimentalFeatures>;
+  /**
+   * Callback called before updating a row with new values in the row and cell editing.
+   * @param {GridRowModel} newRow Row object with the new values.
+   * @param {GridRowModel} oldRow Row object with the old values.
+   * @returns {Promise<GridRowModel>} A promise which resolves with the final values to update the row.
+   */
+  processRowUpdate?: (newRow: GridRowModel, oldRow: GridRowModel) => Promise<GridRowModel>;
```

Comment:
```suggestion
  processRowUpdate?: (newRow: GridRowModel, oldRow: GridRowModel) => GridRowModel | Promise<GridRowModel>;
```

If you remove the `async` on https://codesandbox.io/s/test-new-editing-h9xbfx?file=/demo.tsx, you have a TS error.
But it does work no ?

Reply:
It works because the call is wrapped by `Promise.resolve`. Should we change the definition to include the non-promise return? If we do we have to also explain in the docs. Currently, I say "when it resolves" because it only accepts a promise, but if we change to also allow to return the object I'll have to say in places "when it resolves, or returns" instead. `processRowUpdate` is meant to save the row in a database. I suppose only a small portion of users will use it to save to some local state.

Reply:
I think it is easy enough to support the sync update.

Reply:
So should I keep as it is or allow `GridRowModel` without a promise?

Reply:
Allow `GridRowModel` without a promise

Reply:
Done